### PR TITLE
Remove migrations and migration gem

### PR DESCRIPTION
**Pull request by bilbof**
These are data migrations; we prefer developers to use temporary rake tasks.

We were also not able to deploy router-api to a new env with these tasks (https://deploy.pink.test.govuk.digital/job/Deploy_App/21/console). We couldn't work out why the gem was raising `Mongo::Error::NoServerAvailable`.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,7 +7,6 @@ gem "rails", "~> 5.2"
 
 gem "mongo", "~> 2.4.3"
 gem "mongoid", "~> 6.2"
-gem "mongoid_rails_migrations", git: "https://github.com/alphagov/mongoid_rails_migrations", branch: "avoid-calling-bundler-require-in-library-code-v1.1.0-plus-mongoid-v5-fix"
```
Comment (issyl0) [inline Gemfile:10]: One fewer dependency on a forked gem with fixes from 2017!

**Comment by bilbof** [comment]
@kevindew thanks, I've updated the PR description.

We tried a deploy of this branch and it succeeded (https://deploy.pink.test.govuk.digital/job/Deploy_App/22/console), but we'll try deploying integration now and see if it's happy.

**Comment by kevindew** [comment]
> @kevindew thanks, I've updated the PR description.

👍 

> We tried a deploy of this branch and it succeeded (https://deploy.pink.test.govuk.digital/job/Deploy_App/22/console), but we'll try deploying integration now and see if it's happy.

Cool - I can see it tries to run the command but nothing errors: 

```
15:39:56   * executing `deploy:migrate'
15:39:56   * executing "cd /data/apps/router-api/releases/20200706143955 && govuk_setenv router-api bundle exec rake RAILS_ENV=production  db:migrate"
15:39:56     servers: ["ip-10-200-5-41.eu-west-1.compute.internal", "ip-10-200-4-178.eu-west-1.compute.internal"]
15:39:56     [ip-10-200-4-178.eu-west-1.compute.internal] executing command
15:39:56     [ip-10-200-5-41.eu-west-1.compute.internal] executing command
15:39:57  ** [out :: ip-10-200-5-41.eu-west-1.compute.internal] I, [2020-07-06T14:39:57.022884 #16611]  INFO -- sentry: ** [Raven] Raven 3.0.0 ready to catch errors
15:39:57  ** [out :: ip-10-200-4-178.eu-west-1.compute.internal] I, [2020-07-06T14:39:57.056100 #32151]  INFO -- sentry: ** [Raven] Raven 3.0.0 ready to catch errors
15:39:57     command finished in 1763ms
```

so seems we should flip https://github.com/alphagov/govuk-app-deployment/blob/23ab46de89d9b696e0ca19e91e21dea09444b69a/router-api/config/deploy.rb#L10 when we have a mo but no dramatic rush.

**Comment by bilbof** [comment]
@kevindew thanks for the heads up, I've put up a PR :)

**Pull request merged**